compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1197,7 +1197,14 @@ object SymDenotations {
      *  is defined in Scala 3 and is neither abstract nor open.
      */
     final def isEffectivelySealed(using Context): Boolean =
-      isOneOf(FinalOrSealed) || isClass && !isOneOf(EffectivelyOpenFlags)
+      isOneOf(FinalOrSealed)
+      || isClass && (!isOneOf(EffectivelyOpenFlags)
+      || isLocalToCompilationUnit)
```

Comment:
The indentation suggests this should be an alternative to the line above, but the parentheses make it part of the previous.

@odersky is the indentation just wrong here, or are the parentheses wrong?

Reply:
Indentation looks wrong. Non-classes are never sealed.

Reply:
I am going to reformat as part of one of my next PRs